File: InterFaceOrientationWithStoryboard/InterFaceOrientationWithStoryboard/ViewController.m
//
//  ViewController.m
//  InterFaceOrientationWithStoryboard
//
//  Created by macbook on 15/05/14.
//  Copyright (c) 2014 macbook. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
    
    popButton = [UIButton buttonWithType:UIButtonTypeCustom];
   
    if (([[UIApplication sharedApplication] statusBarOrientation] == UIInterfaceOrientationLandscapeLeft) || ([[UIApplication sharedApplication] statusBarOrientation] == UIInterfaceOrientationLandscapeRight))
    {
        popButton.frame = CGRectMake(980, 10, 30, 30);
        
        topBarView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 1024, 40)];
        topBarView.backgroundColor = [UIColor colorWithPatternImage:[UIImage imageNamed:@"topbarlandscape.png"]];

    }
    else
    {
        //Potrait
        
        popButton.frame = CGRectMake(720, 10, 30, 30);
        
        topBarView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 768, 40)];
        topBarView.backgroundColor = [UIColor colorWithPatternImage:[UIImage imageNamed:@"topbarpotrait.png"]];

    }
    
    [popButton setTitle:@"P" forState:UIControlStateNormal];
    [popButton setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
    [popButton addTarget:self action:@selector(popClicked:) forControlEvents:UIControlEventTouchUpInside];
    [topBarView addSubview:popButton];
    
    [self.view addSubview:topBarView];
    
}

-(void)popClicked:(UIButton *)sender
{
    PopOverViewController *PopoverView =[[PopOverViewController alloc] initWithNibName:@"PopOverViewController" bundle:nil];
    
    self.popOver =[[UIPopoverController alloc] initWithContentViewController:PopoverView];
    self.popOver.popoverContentSize = CGSizeMake(290, 730);
    
    [self showPop:sender];
}

-(void)showPop:(UIButton *)sender
{
    [self.popOver presentPopoverFromRect:sender.frame inView:self.view permittedArrowDirections:UIPopoverArrowDirectionAny animated:YES];
}


- (void)willRotateToInterfaceOrientation:(UIInterfaceOrientation)toInterfaceOrientation duration:(NSTimeInterval)duration
{
    
    if (toInterfaceOrientation == UIInterfaceOrientationLandscapeLeft || toInterfaceOrientation == UIInterfaceOrientationLandscapeRight) {

        popButton.frame = CGRectMake(980, 5, 30, 30);
        
        topBarView.frame = CGRectMake(0, 0, 1024, 40);
        
    }
    else
    {
        popButton.frame = CGRectMake(720, 5, 30, 30);
        
        topBarView.frame = CGRectMake(0, 0, 768, 40);
        
    }
    
}

-(void)didRotateFromInterfaceOrientation:(UIInterfaceOrientation)fromInterfaceOrientation
{
    [self showPop:popButton];
    
    if (([[UIApplication sharedApplication] statusBarOrientation] == UIInterfaceOrientationLandscapeLeft) || ([[UIApplication sharedApplication] statusBarOrientation] == UIInterfaceOrientationLandscapeRight))
    {
        topBarView.backgroundColor = [UIColor colorWithPatternImage:[UIImage imageNamed:@"topbarlandscape.png"]];
    }
    else
    {
        topBarView.backgroundColor = [UIColor colorWithPatternImage:[UIImage imageNamed:@"topbarpotrait.png"]];
    }
    
}


- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: Actionsheet in Swift/Actionsheet in Swift/ViewController.swift
//
//  ViewController.swift
//  Actionsheet in Swift
//
//  Created by Venu on 9/3/15.
//  Copyright (c) 2015 Venu. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        
        let button   = UIButton.buttonWithType(UIButtonType.System) as! UIButton
        button.frame = CGRectMake(100, 100, 100, 50)
        button.backgroundColor = UIColor.greenColor()
        button.setTitle("Test Button", forState: UIControlState.Normal)
        button.addTarget(self, action: "buttonAction:", forControlEvents: UIControlEvents.TouchUpInside)
        
        self.view.addSubview(button)
    }
    
    func buttonAction(sender:UIButton!)
    {
        var title = ""
        var message = "\n\n\n\n\n\n\n\n\n\n";
        var alert = UIAlertController(title: title, message: message, preferredStyle: UIAlertControllerStyle.ActionSheet);
        alert.modalInPopover = true;
        
        //Create the toolbar view - the view witch will hold our 2 buttons
        var toolFrame = CGRectMake(10, 5, 340, 145);
        var toolView: UIView = UIView(frame: toolFrame);
        toolView.backgroundColor = UIColor.redColor()
        
        //add buttons to the view
        var buttonCancelFrame: CGRect = CGRectMake(0, 7, 100, 30); //size & position of the button as placed on the toolView
        
        //Create the cancel button & set its title
        var buttonCancel: UIButton = UIButton(frame: buttonCancelFrame);
        buttonCancel.setTitle("Cancel", forState: UIControlState.Normal);
        buttonCancel.setTitleColor(UIColor.blueColor(), forState: UIControlState.Normal);
        toolView.addSubview(buttonCancel); //add it to the toolView
        
        //Add the target - target, function to call, the event witch will trigger the function call
        buttonCancel.addTarget(self, action: "cancelSelection:", forControlEvents: UIControlEvents.TouchDown);
        
        
        //add buttons to the view
        var buttonOkFrame: CGRect = CGRectMake(170, 7, 100, 30); //size & position of the button as placed on the toolView
        
        //Create the Select button & set the title
        var buttonOk: UIButton = UIButton(frame: buttonOkFrame);
        buttonOk.setTitle("Select", forState: UIControlState.Normal);
        buttonOk.setTitleColor(UIColor.blueColor(), forState: UIControlState.Normal);
        buttonOk.addTarget(self, action: "saveUser:", forControlEvents: UIControlEvents.TouchDown);
        toolView.addSubview(buttonOk); //add to the subview
        
        //add the toolbar to the alert controller
        alert.view.addSubview(toolView);
        
        self.presentViewController(alert, animated: true, completion: nil);
    }
    
    
    func saveUser(sender: UIButton){
        // Your code when select button is tapped
        
        println("User")
    }
    
    func cancelSelection(sender: UIButton){
        println("Cancel");
        self.dismissViewControllerAnimated(true, completion: nil);
        // We dismiss the alert. Here you can add your additional code to execute when cancel is pressed
    }


    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

